src/SingleLineTextEdit.h

```diff
@@ -30,6 +30,9 @@ class SingleLineTextEdit : public QTextEdit
     void setHighlightingEnabled(bool enabled);
     void setTheme(const QString&);
     void rehighlight();
+    bool textEdited();
+Q_SIGNALS:
```

Comment:
We don't explicitly use this QT macro as we are not using any third party libraries that don't understand `signals` - also (and this is a bit of a personal opinion) I like to see signals and slots named as `signal_Xxxxx` and `slot_Xxxxx`respectively - which helps to pick them out in our code base as those things that WE have created and in the case of slots can also help us detect if it is only being called directly in code and is not being `QObject::connect(...)`ed anywhere.

This is because as code is revised later on it might be that a slot method no longer gets called as a slot and can be moved out of a `(public|protected|private) slots:` part of the header file.

It is also worth putting a blank line or two above this line to separate the parts in the header file:
```suggestion


signals:
```